resources/services/apigateway/rest_apis.go

```diff
@@ -924,6 +927,11 @@ func resolveApigatewayRestAPIModelModelTemplate(ctx context.Context, meta schema
 	response, err := svc.GetModelTemplate(ctx, &config, func(options *apigateway.Options) {
 		options.Region = client.Region
 	})
+
+	if client.IsNotFoundError(err) {
```

Comment:
This will panic in `errors.As` inside the fn, move the check inside `if err != nil` :)